src/transformers/models/idefics2/modeling_idefics2.py

```diff
@@ -1626,7 +1626,14 @@ def forward(
             labels = labels.to(logits.device)
             # Shift so that tokens < n predict n
             if attention_mask is not None:
-                shift_attention_mask = attention_mask[..., 1:].to(logits.device)
+                # we use the input attention mask to shift the logits and labels, because it is 2D.
+                # we also crop attn mask in case it is longer, which happens in PrefixTuning with peft
+                shift_attention_mask = attention_mask[:, -(logits.shape[1] - 1) :].to(logits.device)
+                logger.warning_once(
+                    "The final logits were masked using attention mask before calculating the loss. "
+                    "This behavior will be removed in v4.48, you should be masking the `labels` with `-100` "
+                    "in data collator before training starts."
+                )
```

Comment:
that's the only thing I am curious about, we are introducing something new that we know we are gonna deprecate. Not sure this makes a lot of sense to me! Would either not add this, or just not progagate to other models than idefics

Reply:
Yes, I don't know how this got introduced as none of our llms mask out labels. We can also not deprecate and stop propagating, yes. I'll take care of it when new models are added

So, in that case we remove deprecation and leave the fix for PEFT tuning

Reply:
> So, in that case we remove deprecation and leave the fix for PEFT tuning

exactly! We already get bashed enough for the platora of warning we produce, let's not add one more! 😉

Reply:
Thanks for addressing the issue in this PR.

> So, in that case we remove deprecation and leave the fix for PEFT tuning

What fix are we talking about here?

Reply:
oh, that's more transformer-side fix. We are talking about attention mask, which gets extra virtual tokens with PEFT so we have to crop them off to match the shape of `labels` for CELoss

Reply:
Got it, thanks, I thought I might have to fix something in PEFT.